localflavor/bg/validators.py

```diff
@@ -39,30 +44,35 @@ def eik_validator(eik):
     """
     error_message = _('EIK/BULSTAT is not valid')
 
-    def get_checksum(weights, digits):
+    def __call__(self, value):
+        try:
+            value = list(map(int, value))
+        except ValueError:
+            raise ValidationError(self.error_message)
+
+        if not (len(value) in [9, 13] and self.check_eik_base(value)):
+            raise ValidationError(self.error_message)
+
+        if len(value) == 13 and not self.check_eik_extra(value):
+            raise ValidationError(self.error_message)
+
+    def get_checksum(self, weights, digits):
         checksum = sum(weight * digit for weight, digit in zip(weights, digits))
         return checksum % 11
 
-    def check_eik_base(eik):
-        checksum = get_checksum(range(1, 9), eik)
+    def check_eik_base(self, eik):
+        checksum = self.get_checksum(range(1, 9), eik)
         if checksum == 10:
-            checksum = get_checksum(range(3, 11), eik)
+            checksum = self.get_checksum(range(3, 11), eik)
         return eik[8] == checksum % 10
 
-    def check_eik_extra(eik):
+    def check_eik_extra(self, eik):
         digits = eik[8:12]
-        checksum = get_checksum((2, 7, 3, 5), digits)
+        checksum = self.get_checksum((2, 7, 3, 5), digits)
         if checksum == 10:
-            checksum = get_checksum((4, 9, 5, 7), digits)
+            checksum = self.get_checksum((4, 9, 5, 7), digits)
         return eik[-1] == checksum % 10
 
-    try:
-        eik = list(map(int, eik))
-    except ValueError:
-        raise ValidationError(error_message)
-
-    if not (len(eik) in [9, 13] and check_eik_base(eik)):
-        raise ValidationError(error_message)
 
-    if len(eik) == 13 and not check_eik_extra(eik):
-        raise ValidationError(error_message)
+eik_validator = EIKValidator()
+egn_validator = EGNValidator()
```

Comment:
Yes I was thinking to deprecate them. There is no point in callable, because usually logic is complex so you have to split into several methods. And even for simple cases think is better to stick to class validators conventionally.